packages/runtime/container-runtime/src/blobManager.ts

```diff
@@ -217,7 +218,18 @@ export class BlobManager {
                 JSON.stringify(Array.from(this.redirectTable.entries()))),
             );
         }
-        return { entries };
+
+        const builder = new SummaryTreeBuilder();
+        for (const entry of entries) {
+            if (entry.type === TreeEntry.Attachment) {
+                const id = entry.value.id;
+                builder.addAttachment(id);
+            } else {
+                throw new Error("Unexpected TreeEntry type");
+            }
+        }
```

Comment:
The code is wrong as there are two types of blobs added - attachment blob and summary blob.
There is no need to have intermediate states (like entries), you can replace code above with this:

```ts
        blobIds.forEach((blobId) => {
            builder.addAttachment(blobId);
        });

        if (this.redirectTable && this.redirectTable.size > 0) {
            builder.addBlob(
                BlobManager.redirectTableBlobName,
                JSON.stringify(Array.from(this.redirectTable.entries())),
            );
        }

        return builder.getSummaryTree();
```

Reply:
Thanks for this, I clearly didn't have a full understanding of this code path. I'll change this as suggested.